core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -839,15 +839,21 @@ boolean deleteFileInternal(long fileId, boolean recursive, boolean replayed, lon
       // TODO(jiri): What should the Alluxio behavior be when a UFS delete operation fails?
       // Currently, it will result in an inconsistency between Alluxio and UFS.
       if (!replayed && delInode.isPersisted()) {
-        // Delete the file in the under file system.
         try {
-          String ufsPath = mMountTable.resolve(mInodeTree.getPath(delInode)).toString();
-          UnderFileSystem ufs = UnderFileSystem.get(ufsPath, MasterContext.getConf());
-          if (!ufs.exists(ufsPath)) {
-            LOG.warn("File does not exist the underfs: {}", ufsPath);
-          } else if (!ufs.delete(ufsPath, true)) {
-            LOG.error("Failed to delete {}", ufsPath);
-            return false;
+          AlluxioURI alluxioUriToDel = mInodeTree.getPath(delInode);
+          // If this is a mount point, we have deleted all the children and can unmount it
+          if (mMountTable.isMountPoint(alluxioUriToDel)) {
```

Comment:
Created https://alluxio.atlassian.net/projects/ALLUXIO/issues/ALLUXIO-1831 which will go to master.